mod/state-transition/pkg/core/validation_deposits.go

```
@@ -75,6 +83,17 @@
 				)
 			}
 		}
+
+		// BeaconKit enforces a cap on the validator set size.
+		// If genesis deposits breaches the cap we return an error.
+		//#nosec:G701 // can't overflow.
+		if uint32(len(deposits)) > sp.cs.GetValidatorSetCap() {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int -> uint32

(gosec)